src/leap/bitmask/services/mail/imapcontroller.py

```diff
@@ -58,16 +58,21 @@ def start_imap_service(self, userid, offline=False):
         """
         logger.debug('Starting imap service')
 
-        self.imap_service, self.imap_port, \
-            self.imap_factory = imap.start_imap_service(
-                self._soledad,
-                self._keymanager,
-                userid=userid,
-                offline=offline)
+        self.imap_port, self.imap_factory = imap.start_imap_service(
+            self._soledad,
+            userid=userid)
+
+        def assign_incoming_mail(incoming_mail):
+            self.incoming_mail = incoming_mail
 
         if offline is False:
             logger.debug("Starting loop")
-            self.imap_service.start_loop()
+            d = imap.start_incoming_mail_service(
+                self._keymanager,
+                self._soledad,
+                self.imap_factory,
+                userid)
+            d.addCallback(assign_incoming_mail)
 
     def stop_imap_service(self, cv):
```

Comment:
I don't know where the offline flag comes from, is it still used, if it has any meaning I guess if we are offline we should not start incoming mail.